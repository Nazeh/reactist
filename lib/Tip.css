.reactist_tip {
  width: 250px;
  padding: 10px 20px;
}
.reactist_tip__container {
  width: 20px;
  height: 20px;
}
.reactist_tip--title {
  margin: 0 0 10px 0;
  font-size: 0.875rem;
  color: #202020;
  font-weight: 700;
  line-height: 1.7;
}
.reactist_tip--message {
  margin: 0;
  font-size: 0.875rem;
  color: #202020;
  font-weight: 400;
  line-height: 1.7;
}

.reactist_dropdown .trigger {
  cursor: pointer;
}
.reactist_dropdown .body {
  border-radius: 3px;
  border: 1px solid #dcdcdc;
  overflow: hidden;
  box-shadow: 0 1px 8px 0 rgba(0, 0, 0, 0.08);
  z-index: 1;
  background-color: #fff;
}
.reactist_dropdown hr {
  border: none;
  height: 1px;
  background-color: #dcdcdc;
  margin: 0 5px;
}
.reactist_dropdown .with_arrow:after,
.reactist_dropdown .with_arrow:before {
  z-index: 1;
  content: '';
  display: block;
  position: absolute;
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 6px;
  right: 14px;
}
.reactist_dropdown .with_arrow:after {
  top: -11px;
  border-color: transparent transparent #fff transparent;
}
.reactist_dropdown .with_arrow:before {
  top: -12px;
  border-color: transparent transparent #dcdcdc transparent;
}
.reactist_dropdown .with_arrow.top:after {
  top: -1px;
  border-color: #fff transparent transparent transparent;
}
.reactist_dropdown .with_arrow.top:before {
  top: -1px;
  right: 13px;
  border-width: 7px;
  border-color: #dcdcdc transparent transparent transparent;
}

