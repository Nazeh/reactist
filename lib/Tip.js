!function(t,e){"object"==typeof exports&&"object"==typeof module?module.exports=e(require("prop-types"),require("react"),require("classnames"),require("react-dom")):"function"==typeof define&&define.amd?define("reactist",["prop-types","react","classnames","react-dom"],e):"object"==typeof exports?exports.reactist=e(require("prop-types"),require("react"),require("classnames"),require("react-dom")):t.reactist=e(t["prop-types"],t.React,t.classnames,t.ReactDOM)}(window,function(t,e,n,r){return function(t){var e={};function n(r){if(e[r])return e[r].exports;var o=e[r]={i:r,l:!1,exports:{}};return t[r].call(o.exports,o,o.exports,n),o.l=!0,o.exports}return n.m=t,n.c=e,n.d=function(t,e,r){n.o(t,e)||Object.defineProperty(t,e,{enumerable:!0,get:r})},n.r=function(t){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(t,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(t,"__esModule",{value:!0})},n.t=function(t,e){if(1&e&&(t=n(t)),8&e)return t;if(4&e&&"object"==typeof t&&t&&t.__esModule)return t;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:t}),2&e&&"string"!=typeof t)for(var o in t)n.d(r,o,function(e){return t[e]}.bind(null,o));return r},n.n=function(t){var e=t&&t.__esModule?function(){return t.default}:function(){return t};return n.d(e,"a",e),e},n.o=function(t,e){return Object.prototype.hasOwnProperty.call(t,e)},n.p="",n(n.s=66)}({0:function(e,n){e.exports=t},1:function(t,n){t.exports=e},11:function(t,e,n){var r=n(12);"string"==typeof r&&(r=[[t.i,r,""]]);var o={hmr:!0,transform:void 0,insertInto:void 0};n(4)(r,o);r.locals&&(t.exports=r.locals)},12:function(t,e,n){(t.exports=n(3)(!1)).push([t.i,".reactist.dropdown .trigger {\n  cursor: pointer;\n}\n.reactist.dropdown .body {\n  border-radius: 3px;\n  border: 1px solid #dcdcdc;\n  overflow: hidden;\n  box-shadow: 0 1px 8px 0 rgba(0, 0, 0, 0.08);\n  z-index: 1;\n  background-color: #fff;\n}\n.reactist.dropdown hr {\n  border: none;\n  height: 1px;\n  background-color: #ececec;\n  margin: 0 5px;\n}\n.reactist.dropdown .with_arrow:after,\n.reactist.dropdown .with_arrow:before {\n  z-index: 1;\n  content: '';\n  display: block;\n  position: absolute;\n  width: 0;\n  height: 0;\n  border-style: solid;\n  border-width: 6px;\n  right: 14px;\n}\n.reactist.dropdown .with_arrow:after {\n  top: -11px;\n  border-color: transparent transparent #fff transparent;\n}\n.reactist.dropdown .with_arrow:before {\n  top: -12px;\n  border-color: transparent transparent #dcdcdc transparent;\n}\n.reactist.dropdown .with_arrow.top:after {\n  top: -1px;\n  border-color: #fff transparent transparent transparent;\n}\n.reactist.dropdown .with_arrow.top:before {\n  top: -1px;\n  right: 13px;\n  border-width: 7px;\n  border-color: #dcdcdc transparent transparent transparent;\n}\n",""])},2:function(t,e){t.exports=n},3:function(t,e){t.exports=function(t){var e=[];return e.toString=function(){return this.map(function(e){var n=function(t,e){var n=t[1]||"",r=t[3];if(!r)return n;if(e&&"function"==typeof btoa){var o=(a=r,"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,"+btoa(unescape(encodeURIComponent(JSON.stringify(a))))+" */"),i=r.sources.map(function(t){return"/*# sourceURL="+r.sourceRoot+t+" */"});return[n].concat(i).concat([o]).join("\n")}var a;return[n].join("\n")}(e,t);return e[2]?"@media "+e[2]+"{"+n+"}":n}).join("")},e.i=function(t,n){"string"==typeof t&&(t=[[null,t,""]]);for(var r={},o=0;o<this.length;o++){var i=this[o][0];"number"==typeof i&&(r[i]=!0)}for(o=0;o<t.length;o++){var a=t[o];"number"==typeof a[0]&&r[a[0]]||(n&&!a[2]?a[2]=n:n&&(a[2]="("+a[2]+") and ("+n+")"),e.push(a))}},e}},4:function(t,e,n){var r,o,i={},a=(r=function(){return window&&document&&document.all&&!window.atob},function(){return void 0===o&&(o=r.apply(this,arguments)),o}),s=function(t){var e={};return function(t,n){if("function"==typeof t)return t();if(void 0===e[t]){var r=function(t,e){return e?e.querySelector(t):document.querySelector(t)}.call(this,t,n);if(window.HTMLIFrameElement&&r instanceof window.HTMLIFrameElement)try{r=r.contentDocument.head}catch(t){r=null}e[t]=r}return e[t]}}(),c=null,l=0,p=[],u=n(5);function f(t,e){for(var n=0;n<t.length;n++){var r=t[n],o=i[r.id];if(o){o.refs++;for(var a=0;a<o.parts.length;a++)o.parts[a](r.parts[a]);for(;a<r.parts.length;a++)o.parts.push(v(r.parts[a],e))}else{var s=[];for(a=0;a<r.parts.length;a++)s.push(v(r.parts[a],e));i[r.id]={id:r.id,refs:1,parts:s}}}}function d(t,e){for(var n=[],r={},o=0;o<t.length;o++){var i=t[o],a=e.base?i[0]+e.base:i[0],s={css:i[1],media:i[2],sourceMap:i[3]};r[a]?r[a].parts.push(s):n.push(r[a]={id:a,parts:[s]})}return n}function h(t,e){var n=s(t.insertInto);if(!n)throw new Error("Couldn't find a style target. This probably means that the value for the 'insertInto' parameter is invalid.");var r=p[p.length-1];if("top"===t.insertAt)r?r.nextSibling?n.insertBefore(e,r.nextSibling):n.appendChild(e):n.insertBefore(e,n.firstChild),p.push(e);else if("bottom"===t.insertAt)n.appendChild(e);else{if("object"!=typeof t.insertAt||!t.insertAt.before)throw new Error("[Style Loader]\n\n Invalid value for parameter 'insertAt' ('options.insertAt') found.\n Must be 'top', 'bottom', or Object.\n (https://github.com/webpack-contrib/style-loader#insertat)\n");var o=s(t.insertAt.before,n);n.insertBefore(e,o)}}function y(t){if(null===t.parentNode)return!1;t.parentNode.removeChild(t);var e=p.indexOf(t);e>=0&&p.splice(e,1)}function m(t){var e=document.createElement("style");if(void 0===t.attrs.type&&(t.attrs.type="text/css"),void 0===t.attrs.nonce){var r=function(){0;return n.nc}();r&&(t.attrs.nonce=r)}return b(e,t.attrs),h(t,e),e}function b(t,e){Object.keys(e).forEach(function(n){t.setAttribute(n,e[n])})}function v(t,e){var n,r,o,i;if(e.transform&&t.css){if(!(i="function"==typeof e.transform?e.transform(t.css):e.transform.default(t.css)))return function(){};t.css=i}if(e.singleton){var a=l++;n=c||(c=m(e)),r=x.bind(null,n,a,!1),o=x.bind(null,n,a,!0)}else t.sourceMap&&"function"==typeof URL&&"function"==typeof URL.createObjectURL&&"function"==typeof URL.revokeObjectURL&&"function"==typeof Blob&&"function"==typeof btoa?(n=function(t){var e=document.createElement("link");return void 0===t.attrs.type&&(t.attrs.type="text/css"),t.attrs.rel="stylesheet",b(e,t.attrs),h(t,e),e}(e),r=function(t,e,n){var r=n.css,o=n.sourceMap,i=void 0===e.convertToAbsoluteUrls&&o;(e.convertToAbsoluteUrls||i)&&(r=u(r));o&&(r+="\n/*# sourceMappingURL=data:application/json;base64,"+btoa(unescape(encodeURIComponent(JSON.stringify(o))))+" */");var a=new Blob([r],{type:"text/css"}),s=t.href;t.href=URL.createObjectURL(a),s&&URL.revokeObjectURL(s)}.bind(null,n,e),o=function(){y(n),n.href&&URL.revokeObjectURL(n.href)}):(n=m(e),r=function(t,e){var n=e.css,r=e.media;r&&t.setAttribute("media",r);if(t.styleSheet)t.styleSheet.cssText=n;else{for(;t.firstChild;)t.removeChild(t.firstChild);t.appendChild(document.createTextNode(n))}}.bind(null,n),o=function(){y(n)});return r(t),function(e){if(e){if(e.css===t.css&&e.media===t.media&&e.sourceMap===t.sourceMap)return;r(t=e)}else o()}}t.exports=function(t,e){if("undefined"!=typeof DEBUG&&DEBUG&&"object"!=typeof document)throw new Error("The style-loader cannot be used in a non-browser environment");(e=e||{}).attrs="object"==typeof e.attrs?e.attrs:{},e.singleton||"boolean"==typeof e.singleton||(e.singleton=a()),e.insertInto||(e.insertInto="head"),e.insertAt||(e.insertAt="bottom");var n=d(t,e);return f(n,e),function(t){for(var r=[],o=0;o<n.length;o++){var a=n[o];(s=i[a.id]).refs--,r.push(s)}t&&f(d(t,e),e);for(o=0;o<r.length;o++){var s;if(0===(s=r[o]).refs){for(var c=0;c<s.parts.length;c++)s.parts[c]();delete i[s.id]}}}};var g,w=(g=[],function(t,e){return g[t]=e,g.filter(Boolean).join("\n")});function x(t,e,n,r){var o=n?"":r.css;if(t.styleSheet)t.styleSheet.cssText=w(e,o);else{var i=document.createTextNode(o),a=t.childNodes;a[e]&&t.removeChild(a[e]),a.length?t.insertBefore(i,a[e]):t.appendChild(i)}}},5:function(t,e){t.exports=function(t){var e="undefined"!=typeof window&&window.location;if(!e)throw new Error("fixUrls requires window.location");if(!t||"string"!=typeof t)return t;var n=e.protocol+"//"+e.host,r=n+e.pathname.replace(/\/[^\/]*$/,"/");return t.replace(/url\s*\(((?:[^)(]|\((?:[^)(]+|\([^)(]*\))*\))*)\)/gi,function(t,e){var o,i=e.trim().replace(/^"(.*)"$/,function(t,e){return e}).replace(/^'(.*)'$/,function(t,e){return e});return/^(#|data:|http:\/\/|https:\/\/|file:\/\/\/|\s*$)/i.test(i)?t:(o=0===i.indexOf("//")?i:0===i.indexOf("/")?n+i:r+i.replace(/^\.\//,""),"url("+JSON.stringify(o)+")")})}},6:function(t,e){t.exports=r},63:function(t,e,n){var r=n(64);"string"==typeof r&&(r=[[t.i,r,""]]);var o={hmr:!0,transform:void 0,insertInto:void 0};n(4)(r,o);r.locals&&(t.exports=r.locals)},64:function(t,e,n){(t.exports=n(3)(!1)).push([t.i,".reactist.tip {\n  width: 250px;\n  padding: 10px 20px;\n}\n.reactist.tip__container {\n  width: 20px;\n  height: 20px;\n}\n.reactist.tip--title {\n  margin: 0 0 10px 0;\n  font-size: 0.875rem;\n  color: #202020;\n  font-weight: 700;\n  line-height: 1.7;\n}\n.reactist.tip--message {\n  margin: 0;\n  font-size: 0.875rem;\n  color: #202020;\n  font-weight: 400;\n  line-height: 1.7;\n}\n",""])},66:function(t,e,n){"use strict";n.r(e);n(63);var r=n(1),o=n.n(r),i=n(0),a=n.n(i),s=n(2),c=n.n(s),l=function(){return o.a.createElement("svg",{width:"18px",height:"18px",viewBox:"0 0 18 18"},o.a.createElement("g",{stroke:"none",strokeWidth:"1",fill:"none",fillRule:"evenodd"},o.a.createElement("g",{fill:"#979797"},o.a.createElement("path",{d:"M9,17 C13.418278,17 17,13.418278 17,9 C17,4.581722 13.418278,1 9,1 C4.581722,1 1,4.581722 1,9 C1,13.418278 4.581722,17 9,17 Z M9,18 C4.02943725,18 0,13.9705627 0,9 C0,4.02943725 4.02943725,0 9,0 C13.9705627,0 18,4.02943725 18,9 C18,13.9705627 13.9705627,18 9,18 Z",fillRule:"nonzero"}),o.a.createElement("path",{d:"M5.625,6.671875 L6.9609375,6.671875 C7.046875,5.5546875 7.796875,4.859375 8.9765625,4.859375 C10.140625,4.859375 10.890625,5.5703125 10.890625,6.5390625 C10.890625,7.375 10.5390625,7.8671875 9.6328125,8.421875 C8.5625,9.0625 8.0625,9.765625 8.0703125,10.796875 L8.0703125,11.515625 L9.4375,11.515625 L9.4375,11.0078125 C9.4375,10.1875 9.71875,9.75 10.7109375,9.1640625 C11.7109375,8.5625 12.34375,7.6953125 12.34375,6.46875 C12.34375,4.859375 11,3.640625 9.03125,3.640625 C6.8359375,3.640625 5.7109375,4.984375 5.625,6.671875 Z M8.859375,15.140625 C9.4921875,15.140625 9.9375,14.6875 9.9375,14.0546875 C9.9375,13.4140625 9.4921875,12.9609375 8.859375,12.9609375 C8.2265625,12.9609375 7.7734375,13.4140625 7.7734375,14.0546875 C7.7734375,14.6875 8.2265625,15.140625 8.859375,15.140625 Z"}))))},p=n(7),u=function(t){var e=t.title,n=t.message,r=t.top,i=t.className,a=c()("reactist tip__container",i);return o.a.createElement(p.default.Box,{top:r,className:a},o.a.createElement(p.default.Trigger,null,o.a.createElement(l,null)),o.a.createElement(p.default.Body,null,o.a.createElement("div",{className:"reactist tip"},o.a.createElement("p",{className:"reactist tip--title"},e),o.a.createElement("p",{className:"reactist tip--message"},n))))};u.displayName="Tip",u.defaultProps={top:!1},u.propTypes={top:a.a.bool,title:a.a.string.isRequired,message:a.a.oneOfType([a.a.string,a.a.arrayOf(a.a.node),a.a.node]).isRequired,className:a.a.string};e.default=u},7:function(t,e,n){"use strict";n.r(e);n(11);var r=n(1),o=n.n(r),i=n(6),a=n.n(i),s=n(0),c=n.n(s),l=n(2),p=n.n(l);function u(t){return(u="function"==typeof Symbol&&"symbol"==typeof Symbol.iterator?function(t){return typeof t}:function(t){return t&&"function"==typeof Symbol&&t.constructor===Symbol&&t!==Symbol.prototype?"symbol":typeof t})(t)}function f(t,e){if(!(t instanceof e))throw new TypeError("Cannot call a class as a function")}function d(t,e){for(var n=0;n<e.length;n++){var r=e[n];r.enumerable=r.enumerable||!1,r.configurable=!0,"value"in r&&(r.writable=!0),Object.defineProperty(t,r.key,r)}}function h(t,e,n){return e&&d(t.prototype,e),n&&d(t,n),t}function y(t,e){return!e||"object"!==u(e)&&"function"!=typeof e?b(t):e}function m(t){return(m=Object.setPrototypeOf?Object.getPrototypeOf:function(t){return t.__proto__||Object.getPrototypeOf(t)})(t)}function b(t){if(void 0===t)throw new ReferenceError("this hasn't been initialised - super() hasn't been called");return t}function v(t,e){if("function"!=typeof e&&null!==e)throw new TypeError("Super expression must either be null or a function");t.prototype=Object.create(e&&e.prototype,{constructor:{value:t,writable:!0,configurable:!0}}),e&&g(t,e)}function g(t,e){return(g=Object.setPrototypeOf||function(t,e){return t.__proto__=e,t})(t,e)}var w=function(t){function e(t,n){var r;return f(this,e),(r=y(this,m(e).call(this,t,n))).state={show_body:!1,top:t.top||!1},r._handleClickOutside=r._handleClickOutside.bind(b(r)),r._setPosition=r._setPosition.bind(b(r)),r._toggleShowBody=r._toggleShowBody.bind(b(r)),r._timeout=null,r}return v(e,o.a.Component),h(e,[{key:"componentWillUnmount",value:function(){document.removeEventListener("click",this._handleClickOutside,!0),this._timeout&&clearTimeout(this._timeout)}},{key:"_handleClickOutside",value:function(t){var e=this;a.a.findDOMNode(this).contains(t.target)?this.props.allowBodyInteractions||(this._timeout=setTimeout(function(){e.state.show_body&&e._toggleShowBody()},100)):this._toggleShowBody()}},{key:"_toggleShowBody",value:function(){this.state.show_body?(this.props.onHideBody&&this.props.onHideBody(),document.removeEventListener("click",this._handleClickOutside,!0)):(this.props.onShowBody&&this.props.onShowBody(),document.addEventListener("click",this._handleClickOutside,!0)),this.setState({show_body:!this.state.show_body})}},{key:"_getTriggerComponent",value:function(){var t=this.props.children[0];return o.a.cloneElement(t,{onClick:this._toggleShowBody})}},{key:"_setPosition",value:function(t){if(t){var e=document.getElementById(this.props.scrolling_parent);if(e){var n=a.a.findDOMNode(this),r=a.a.findDOMNode(this).offsetTop,o=n.querySelector(".trigger").clientHeight,i=t.clientHeight,s=e.clientHeight+e.scrollTop-r-o<i;s!==this.state.top&&this.setState({top:s})}}}},{key:"_getBodyComponent",value:function(){var t=this.state.top,e=this.props,n=e.right,r=void 0!==n&&n,i=e.children[1],a={top:t,right:r,setPosition:this._setPosition},s=p()({body_wrapper:!0,with_arrow:!0,top:t,bottom:!t});return!!this.state.show_body&&o.a.createElement("div",{className:s,style:{position:"relative"}},o.a.cloneElement(i,a))}},{key:"render",value:function(){var t=p()("reactist dropdown",this.props.className),e=this.state.top;return o.a.createElement("div",{style:{display:"inline-block"},className:t},e&&this._getBodyComponent(),this._getTriggerComponent(),!e&&this._getBodyComponent())}}]),e}();w.displayName="Dropdown.Box",w.propTypes={top:c.a.bool,right:c.a.bool,scrolling_parent:c.a.string,allowBodyInteractions:c.a.bool,onShowBody:c.a.func,onHideBody:c.a.func,className:c.a.string,children:c.a.any};var x=function(t){function e(t,n){var r;return f(this,e),(r=y(this,m(e).call(this,t,n)))._onClick=r._onClick.bind(b(r)),r}return v(e,o.a.Component),h(e,[{key:"_onClick",value:function(t){t.preventDefault(),t.stopPropagation(),this.props.onClick(t)}},{key:"render",value:function(){return o.a.createElement("div",{style:{display:"block"},className:"trigger",onClick:this._onClick},this.props.children)}}]),e}();x.displayName="Dropdown.Trigger",x.propTypes={onClick:c.a.func,children:c.a.any};var _=function(t){function e(){return f(this,e),y(this,m(e).apply(this,arguments))}return v(e,o.a.Component),h(e,[{key:"render",value:function(){var t={position:"absolute",right:0,top:0};return this.props.top&&(t.top="auto",t.bottom=0),this.props.right&&(t.right="auto",t.left=0),o.a.createElement("div",{ref:this.props.setPosition,style:t,className:"body",id:"reactist-dropdown-body"},this.props.children)}}]),e}();_.displayName="Dropdown.Body",_.propTypes={top:c.a.bool,right:c.a.bool,setPosition:c.a.func,children:c.a.any},e.default={Box:w,Trigger:x,Body:_}}})});
//# sourceMappingURL=Tip.map