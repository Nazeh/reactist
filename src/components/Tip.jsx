import './styles/tip.less'

import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import ThinQuestionMark from './icons/ThinQuestionMarkIcon.svg'
import Dropdown from './Dropdown'

const Tip = ({ title, message, top, className }) => {
    const tipClass = classNames('reactist tip__container', className)
    return (
        <Dropdown.Box top={top} className={tipClass}>
            <Dropdown.Trigger>
                <ThinQuestionMark />
            </Dropdown.Trigger>
            <Dropdown.Body>
                <div className="reactist tip">
                    <p className="reactist tip--title">{title}</p>
                    <p className="reactist tip--message">{message}</p>
                </div>
            </Dropdown.Body>
        </Dropdown.Box>
    )
}
Tip.displayName = 'Tip'
Tip.defaultProps = {
    top: false
}
Tip.propTypes = {
    /** Whether the tip content should be displayed to the top or not. Maps to the Dropdown.Box top property. */
    top: PropTypes.bool,
    /** Title of the tip. */
    title: PropTypes.string.isRequired,
    /** Message of the tip. */
    message: PropTypes.string.isRequired,
    /** Addtional css class that is applied to the Tip. */
    classname: PropTypes.string
}

export default Tip
