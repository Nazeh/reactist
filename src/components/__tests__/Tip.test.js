import React from 'react'
import { shallow } from 'enzyme'

import Tip from '../Tip'

describe('Tip', () => {
    it('renders without crashing', () => {
        const tip = shallow(getTip())
        expect(tip).toMatchSnapshot()
    })

    // Helpers ================================================================
    const getTip = props => (
        <Tip title="Title of the Tip" message="Message of the Tip" {...props} />
    )
})
