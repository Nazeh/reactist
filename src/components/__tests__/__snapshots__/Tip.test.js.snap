// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tip renders without crashing 1`] = `
<Dropdown.Box
  className="reactist_tip__container"
  top={false}
>
  <Dropdown.Trigger>
    <ThinQuestionMark />
  </Dropdown.Trigger>
  <Dropdown.Body>
    <div
      className="reactist_tip"
    >
      <p
        className="reactist_tip--title"
      >
        Title of the Tip
      </p>
      <p
        className="reactist_tip--message"
      >
        Message of the Tip
      </p>
    </div>
  </Dropdown.Body>
</Dropdown.Box>
`;
